### DISCUSSION TITLE
Can't make prerender work when a (node_)module imports JSON  file

### DISCUSSION THREAD

**Discussion by Xvdfe** [preactjs/wmr#793] [Q&A] [answered]
I'm trying to prerender an app and add Twind compiled shims to the mix (would love to use Hoofd too later).

Following the samples provided by the WMR docs as well as those from Twind I ended up adding a `prerender.js` file to my `public` directory:

```javascript
import { prerender as ssr } from 'preact-iso'
import 'twind/shim/server'

export default async (app, options = {}) => {
  const result = await ssr(app)
  result.html = `${result.html}`
  return result
}
```

As you can see there's no real use of Twind shim in the source code yet, only an import.

My `prerender` function in `index.js` looks like:

```javascript
export async function prerender(data) {
  const { default: prerender } = await import('./prerender')
  return prerender(<App {...data} />, { shim: true })
}
```

Everything works well with `wmr serve` / `npm start`. Using `wmr build --prerender` / `npm run build` however generates an error:

```console
Error: Could not load json:npm/entities@2.2.0/lib/maps/entities.json (imported by npm/htmlparser2@6.1.0/lib/Tokenizer.js): ENOENT: no such file or directory, open '/Users/raphael/src/ceptem/ceptem-website/public/npm/entities@2.2.0/lib/maps/entities.json'
```

Or same error with `legacy.json` or same error with `xml.json`, all files that I can find at `node_modules/entities/lib/maps`, the fact that the error changes depending upon IDontKnowWhat makes me think about an asynchronous issue...

`prerender.js` is imported asynchronously.

The import chain is as follow:
- `index.js`:
   ```
   export async function prerender(data) {
     const { default: prerender } = await import('./prerender')
   ```
- `prerender.js`:
   ```
   import 'twind/shim/server'
   ```
- `node_modules/twind/shim/server/server.js`:
   ```
   import htmlparser2Tokenizer from "htmlparser2/lib/Tokenizer";
   ```
- `node_modules/htmlparser2/lib/Tokenizer.js`:
   ```
   var entities_json_1 = __importDefault(require("entities/lib/maps/entities.json"));
   var legacy_json_1 = __importDefault(require("entities/lib/maps/legacy.json"));
   var xml_json_1 = __importDefault(require("entities/lib/maps/xml.json"));
   ```

And, as said, these files exist.

Now, I don't post this as an issue as I think it's coming from an incorrect usage of mine, does anyone has any idea?

**Comment by rschristian** [comment, marked as answer]
I'll need to take a look at the errors myself a bit later, but [Twind does offer a WMR integration](https://github.com/tw-in-js/use-twind-with/tree/main/packages/wmr), should work right out of the box for you.

**Reply by Xvdfe** [reply]
How did I not find this but [this](https://github.com/tw-in-js/example-wmr/blob/7e33313847226a3659ce0316b4dd5aaf7c7f39ed/public/index.js) (which was only a bit outdated) ?

Thanks a lot, the build error message disappeared, seems the `hidden` attribute handling is different but I'm confident I can make it work now.

**Reply by rschristian** [reply]
Do you know what linked to that version of the repo? The head of the repo is up-to-date, in that it uses the `@twind/wmr` package. Wondering if some docs point at the older version somewhere.

**Reply by Xvdfe** [reply]
Yes, [this issue discussion](https://github.com/preactjs/wmr/issues/250), as at first I was looking to integrate vanilla Tailwind with WMR.

**Reply by rschristian** [reply]
Thanks, I'll add a footnote.
